src/types/new_type_row.rs

```diff
@@ -88,6 +90,23 @@ impl<T: TypeClass> TypeRow<T> {
                 .collect::<Vec<Type<T2>>>(),
         )
     }
+
+    /// Return the type row of variants required to define a Sum of Tuples type
+    /// given the rows of each tuple
+    pub fn predicate_variants_row(variant_rows: impl IntoIterator<Item = TypeRow<T>>) -> Self {
+        variant_rows
+            .into_iter()
+            .map(Type::new_tuple)
+            .collect_vec()
+            .into()
+    }
+
+    /// Returns the smallest [TypeTag] that contains all elements of the row
```

Comment:
or, erm, was it `actual_tag`?? (I do not insist, having a different name is reasonable)